contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,401 @@
+    bool private _stakingPaused;
+
+    // stake params
+    uint256 public minSelfDelegation;
+    uint256 public minDelegationChange;
+    uint256 public maxElectedValidators;
+
+    // slash params
+    uint256 public downtimeSlashAmount;
+    uint256 public doubleSignSlashAmount;
+    uint256 public downtimeJailTime;
+    uint256 public doubleSignJailTime;
+
+    // validator address => validator info
+    mapping(address => Validator) public validators;
+    // validator vote address => validator address
+    mapping(bytes => address) public voteAddressToValidator;
+    // slash key => slash record
+    mapping(bytes32 => SlashRecord) public slashRecords;
+
+    uint256 public totalPooledBNB;
+
+    IBSCValidatorSet.Validator[] public eligibleValidators;
+    uint256[] public eligibleValidatorDelegatedAmounts;
+    bytes[] public eligibleValidatorVoteAddrs;
+
+    struct Validator {
+        address consensusAddress;
+        bool jailed;
+        address poolModule; // staking pool
+        uint256 totalPooledBNB;
+        uint256 totalShares;
+        bytes voteAddress;
+        Description description;
+        Commission commission;
+        uint256[20] slots;
+    }
+
+    struct Description {
+        string moniker;
+        string identity;
+        string website;
+        string details;
+    }
+
+    struct Commission {
+        uint256 rate; // the commission rate charged to delegators
+        uint256 maxRate; // maximum commission rate which validator can ever charge
+        uint256 maxChangeRate; // maximum daily increase of the validator commission
+    }
+    //uint256 updateTime;
+
+    enum SlashType {
+        DoubleSign,
+        DownTime,
+        MaliciousVote
+    }
+
+    struct SlashRecord {
+        uint256 slashAmount;
+        uint256 slashHeight;
+        uint256 jailUntil;
+        SlashType slashType;
+    }
+
+    /*----------------- modifiers -----------------*/
+    modifier onlyInitialized() {
+        require(_initialized != 0, "NOT_INITIALIZED");
+        _;
+    }
+
+    modifier validatorExist(address validator) {
+        require(validators[validator].poolModule != address(0), "VALIDATOR_NOT_EXIST");
+        _;
+    }
+
+    /*----------------- init -----------------*/
+    function initialize() public initializer {
+        minSelfDelegation = INIT_MIN_SELF_DELEGATION;
+        minDelegationChange = INIT_MIN_DELEGATION_CHANGE;
+        maxElectedValidators = INIT_MAX_ELECTED_VALIDATORS;
+        downtimeSlashAmount = INIT_DOWNTIME_SLASH_AMOUNT;
+        doubleSignSlashAmount = INIT_DOUBLE_SIGN_SLASH_AMOUNT;
+        downtimeJailTime = INIT_DOWNTIME_JAIL_TIME;
+        doubleSignJailTime = INIT_DOUBLE_SIGN_JAIL_TIME;
+
+        _initialized = 1;
+    }
+
+    /*----------------- external functions -----------------*/
+    function createValidator(bytes calldata voteAddress, bytes calldata blsProof, Commission calldata commission, Description calldata description) external payable onlyInitialized {
```

Comment:
what about the consensus address?

Reply:
I think we need an operator address too, who is allowed to edit valdiator

Reply:
Done